pwndbg/search.py

```diff
@@ -18,18 +16,14 @@
 import pwndbg.vmmap
 
 
-def search(searchfor, mapping=None, start=None, end=None, 
+def search(searchfor, mapping=None, pages=None, start=None, end=None,
            executable=False, writable=False):
-    value = searchfor
-    size  = None
-
     i = gdb.selected_inferior()
 
-    maps = pwndbg.vmmap.get()
-    hits = []
+    maps = pwndbg.vmmap.get() if not pages else pages
 
     if end and start:
-        maps = [m for m in maps if start <= m < end]
+        maps = [m for m in maps if start >= m.start and end < m.end]
```

Comment:
This reverses the old logic, by requiring that `start` and `end` be entirely within a given mapping, rather than `start` and `end` containing a given mapping.